ghcup-prereleases-0.0.7.yaml

```diff
@@ -540,6 +540,92 @@ ghcupDownloads:
       viTags:
       - Prerelease
       - base-4.18.0.0
+    9.6.0.20230128:
+      viArch:
+        A_32:
+          Linux_Debian:
+            <10: &ghc-961alpha2-64-deb9
+              dlHash: e2f3e622f1aecfe0b6a305d0fb997e83453ecbc2949cb2b393549e35f2b062e1
+              dlSubdir: ghc-9.6.0.20230128-i386-unknown-linux
+              dlUri: https://downloads.haskell.org/~ghc/9.6.0.20230128/ghc-9.6.0.20230128-i386-deb9-linux.tar.xz
+            unknown_versioning: *ghc-961alpha2-64-deb9
+          Linux_Mint:
+            unknown_versioning: *ghc-961alpha2-64-deb9
+          Linux_Ubuntu:
+            unknown_versioning: *ghc-961alpha2-64-deb9
+          Linux_UnknownLinux:
+            unknown_versioning: *ghc-961alpha2-64-deb9
+        A_64:
+          Darwin:
+            unknown_versioning:
+              dlHash: 694d70b648853da04b115d7cdd4fb9bd953fb6361638ee83dda1f7fc8fb344f7
+              dlSubdir: ghc-9.6.0.20230128-x86_64-apple-darwin
+              dlUri: https://downloads.haskell.org/~ghc/9.6.0.20230128/ghc-9.6.0.20230128-x86_64-apple-darwin.tar.xz
+          Linux_Alpine:
+            unknown_versioning:
+              dlHash: 19d8d59a6d68c3b3174a8e3513d3b9236ec8f505428e7e909af4e7b108716416
+              dlSubdir: ghc-9.6.0.20230128-x86_64-unknown-linux
+              dlUri: https://downloads.haskell.org/~ghc/9.6.0.20230128/ghc-9.6.0.20230128-x86_64-alpine3_12-linux-static-int_native.tar.xz
```

Comment:
Last I tried this was unusable for me and caused linking errors. Don't we have a dynamically linked alpine bindist? And if not, why?

Reply:
My initial motivation for introducing Alpine bindists was to facilitate users who wanted a solution for static-linking. We can introduce a dynamic job for the next pre-release if this would be helpful. Was [this](https://gitlab.haskell.org/ghc/ghc/-/issues/22237) the issue you observed with the static bindist?

Reply:
No, it was this https://github.com/haskell/haskell-language-server/actions/runs/3719591446/jobs/6308547604

Invocation:

```
cabal --store-dir=/__w/haskell-language-server/haskell-language-server/store v2-install --project-file="cabal.project" -w "ghc-8.10.7" --disable-tests --disable-profiling -O2 --overwrite-policy=always --install-method=copy --installdir="/__w/haskell-language-server/haskell-language-server/out/8.10.7" exe:haskell-language-server exe:haskell-language-server-wrapper
```

Errors:

```
2022-12-17T11:03:04.3496562Z [1 of 1] Compiling Main             ( exe/Main.hs, dist/build/haskell-language-server/haskell-language-server-tmp/Main.o )
2022-12-17T11:03:04.3496902Z 
2022-12-17T11:03:04.3497009Z exe/Main.hs:6:8: error:
2022-12-17T11:03:04.3497397Z     Could not find module ‘Prelude’
2022-12-17T11:03:04.3497895Z     Perhaps you haven't installed the "dyn" libraries for package ‘base-4.16.4.0’?
2022-12-17T11:03:04.3498423Z     Use -v (or `:set -v` in ghci) to see a list of the files searched for.
2022-12-17T11:03:04.3504580Z   |
2022-12-17T11:03:04.3504922Z 6 | module Main(main) where
2022-12-17T11:03:04.3505158Z   |        ^^^^
```

Reply:
As far as I can tell, this isn't a linking failure but rather a `cabal-install` configuration issue. Specifically, `cabal-install` is trying to build a dynamic executable despite the fact that we don't have dynamic libraries. Teaching `cabal-install` to handle this automatically is indeed a bit tricky.

Reply:
Does these current bindist GHCs allow using TH, doesn't that require dynamic linking?